apps/scripts/AliveTest.c

```diff
@@ -0,0 +1,143 @@
+// 4KB of D_MEM
+// 0x400800 - 0x400fff - Shared
+//
+// 0x400600 - 0x400800 - Thread 3
+// 0x400400 - 0x400600 - Thread 2
+// 0x400200 - 0x400400 - Thread 1
+// 0x400000 - 0x400200 - Thread 0
+
+// REGION == 2'b01;
+
+#define D_MEM_SHARED ((volatile int *) (0x00400f00))
+
+//////////Register-Immediate Instructions///////////
+void ADDI(int a){
+    D_MEM_SHARED[0] = a+1;
+}
+void SLTI(int a){
+    D_MEM_SHARED[1] = a<5;
+}
+void ANDI(int a){
+    D_MEM_SHARED[2] = a&6;
+}
+void ORI(int a){
+    a=a|26;
+    if(a==30)
+    	D_MEM_SHARED[3] = 1;
+}
+void XORI(int a){
+    D_MEM_SHARED[4] = a^42;
+}
+void SLLI(int a){
+    a=a<<3;
+    if(a==64)
+    	D_MEM_SHARED[5] = 1;
+}
+void SRLI(int a){
```

Comment:
if you want to force a logical shift without writing complicated assembly, you can declare `a` as unsigned int